src/hooks/useOnEscape.test.tsx

```diff
@@ -0,0 +1,11 @@
+import { renderHook } from "@testing-library/react-hooks";
+import { fireEvent } from "@testing-library/dom";
+
+import { useOnEscape } from "./useOnEscape";
+
+it("calls the callback when the escape key is pressed", () => {
+  const onEscape = jest.fn();
+  renderHook(() => useOnEscape(onEscape));
+  fireEvent.keyDown(document, { key: "Escape", code: "Escape" });
```

Comment:
In a real world a whole sequence of keyboard events would be fired. It'd be better to not focus on implementation details though. A more user-centric way of testing this would be to use `userEvent.keyboard("{esc}")` instead.

Reply:
For some reason when I tried that the events wouldn't fire. Do you happen to know if you have to get it to target the document somehow so the events fire? I wondered if it was an issue with the `react-hooks` test lib.

Reply:
I tested this locally and had no issues: https://github.com/huwshimi/vanilla-react-components/compare/navigation-component...petermakowski:useOnEscape-userEvent

Reply:
Oh great thanks. Not sure what I was doing before, but that seems to work now!